### PR TITLE
[LD-52] Codebase improving - domain package

### PR DESCRIPTION
**Pull request by Krzysiudan**
https://loudius.atlassian.net/jira/software/projects/LD/boards/3

## Changes
- moved `AuthRepositoryImpl` into the file with `AuthRepository` interface
- moved repositories into the additional package - repository. 

Feel free to question mine refactor.

### CODE REVIEW

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 package com.appunite.loudius.di
```
Comment (kezc) [inline app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: thought: we've left a lonely `UserLocalDataSource` in the domain package. Maybe we could create a separate package for that? WDYT?

Comment (nowakweronika) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: Good idea :D

**Review by nowakweronika — COMMENTED** [review]
conflict

File: app/src/main/java/com/appunite/loudius/di/GithubModule.kt (hunk 1 above)
Comment (Krzysiudan) [reply app/src/main/java/com/appunite/loudius/di/GithubModule.kt]: Yeah, good idea 👌 
Moved that to the `store` package.

**Review by nowakweronika — APPROVED** [review]

**Review by kezc — APPROVED** [review]

**Pull request merged**